flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputProcessor.java

```diff
@@ -293,9 +291,12 @@ private void updateAvailability(InputStatus status, StreamTaskInput input) {
 	}
 
 	private void checkAndSetAvailable(int inputIndex) {
-		StreamTaskInput input = getInput(inputIndex);
+		CompletableFuture<?> inputAvailable = getInput(inputIndex).isAvailable();
 		InputStatus status = (inputIndex == 0 ? firstInputStatus : secondInputStatus);
```

Comment:
A bit unrelated, but do you have any idea, why we are not using an array of InputStatus? Feels like this check is an unnecessary pipeline breaker.

Reply:
Performance reasons (arrays would add extra overhead), however I think It was never measured in the flink context (form other projects I know that this can be an issue and my best guess would it be that's the same here).